### PR TITLE
Wrong results in is_isotopic method of the Link class for certain chiral link

### PR DESCRIPTION
**Pull request by soehms**
The following result is wrong, since `K10_67` is a chiral knot:

```
sage: L = KnotInfo.K10_67
....: L1 = L.link()
....: L1r = L.link().reverse()
....: L1.is_isotopic(L1r)
True
```

The problem is that the `get_knotinfo` method currently does not distinguish between all four symmetry mutants of a chiral knot or link. In the example, `get_knotinfo` detects both knots as *not mirrored* to the symmetry mutant of `K10_67` recorded in the KnotInfo database, as you can see from the verbose messages:

```
sage: set_verbose(1)
sage: L1.is_isotopic(L1r)
verbose 1 (4365: link.py, is_isotopic) KnotInfo other: KnotInfo.K10_67 mirrored False
True
```

The wrong conclusion in `is_isotopic` is that they must be isotopic.

The idea in this PR is to replace the boolean `mirrored` in the output of `get_knotinfo` with the value of an enum representing all symmetry mutants for an entry of the KnotInfo database. Of course, this requires major revisions in the `get_knotinfo` and `is_isotopic` methods. The key to this is a new local method `_knotinfo_matching_dict` that builds on `_knotinfo_matching_list` and returns the matching list for all symmetry mutants.

On the occasion of that revision, I also fix other wrong results for example this one of a multi-component link which is not amphicheiral, but reversible:

```
sage: L = KnotInfo.L6a2_0
sage: L1 = L.link()
sage: L2 = L.link(L.items.braid_notation)
sage: L1.is_isotopic(L2)
False
```

This bug is due to `get_knotinfo` returning the wrong mirror version for `L1` concerning `L6a2_1`:

```
...
sage: L1.get_knotinfo(unique=False)
[(<KnotInfo.L6a2_0: 'L6a2{0}'>, False), (<KnotInfo.L6a2_1: 'L6a2{1}'>, False)]
sage: L2.get_knotinfo(unique=False)
[(<KnotInfo.L6a2_0: 'L6a2{0}'>, False), (<KnotInfo.L6a2_1: 'L6a2{1}'>, True)]
```

In addition to these fixes, improvements and corresponding adjustments, I correct some document formatting.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by soehms** [comment]
I add some more tests. But since most lines of code that appear in the Codecov messages require the optional `database_knotinfo` package, I think this won't improve the score much.

File: src/sage/knots/knotinfo.py [hunk 1]
```diff
@@ -335,6 +335,45 @@ def knotinfo_bool(string):
...
+            sage: from sage.knots.knotinfo import SymmetryMutant
+            sage: sorted(SymmetryMutant)        # indirect doctest
+            [<SymmetryMutant.mixed: 'x'>,
+            <SymmetryMutant.itself: 's'>,
+            <SymmetryMutant.reverse: 'r'>,
+            <SymmetryMutant.concordance_inverse: 'mr'>,
+            <SymmetryMutant.mirror_image: 'm'>,
+            <SymmetryMutant.unknown: '?'>]
+        """
+        return self.value < other.value
```
Comment (tscrim) [inline src/sage/knots/knotinfo.py:376]: This is a `__gt__` method but the comparison is `lt`. Is this intentional?

File: src/sage/knots/link.py [hunk 2]
```diff
@@ -4246,33 +4328,45 @@ def answer(L):
...
-                    # polynomial does not distinguish mirror images (see the above
-                    # example ``k11m``).
-                    if unique:
-                        raise NotImplementedError('mirror type of this link cannot be uniquely determined%s' % non_unique_hint)
-                    mirrored = '?'
-                elif L in lm:
-                    mirrored = True
-                else:
-                    mirrored = False
+                for k in match_lists.keys():
```
Comment (tscrim) [inline src/sage/knots/link.py]:
```suggestion
                for k in match_lists:
```

File: src/sage/knots/link.py [hunk 3]
```diff
@@ -4246,33 +4328,45 @@ def answer(L):
...
+                    lk = match_lists[k]
+                    if proves[k] and L in lk:
+                        lk.remove(L)
+                        sym_mut = k
+                        break
+
+            if not sym_mut:
+                for k in match_lists.keys():
```
Comment (tscrim) [inline src/sage/knots/link.py]:
```suggestion
                for k in match_lists:
```

File: src/sage/knots/link.py [hunk 4]
```diff
@@ -4282,31 +4376,28 @@ def answer_unori(S):
             if not mirror_version:
                 return S
 
-            mirrored = [answer(L)[1] for L in S]
-            if all(mirrored):
+            sym_mut = [answer(L)[1] for L in S]
+            if all(i is SymmetryMutant.mirror_image  for i in sym_mut):
```
Comment (tscrim) [inline src/sage/knots/link.py]:
```suggestion
            if all(i is SymmetryMutant.mirror_image for i in sym_mut):
```

File: src/sage/knots/link.py [hunk 5]
```diff
@@ -4282,31 +4376,28 @@ def answer_unori(S):
...
+            return S, SymmetryMutant.mixed
 
         def answer_list(l):
             r"""
             Return a list of items of the KnotInfo database according to the keyword
             argument ``unique``.
             """
             if not unique:
-                return sorted([answer(L) for L in l])
+                return sorted(set([answer(L) for L in l]))
```
Comment (tscrim) [inline src/sage/knots/link.py:4398]: This seems to be making the output have unique instances. Perhaps I am misunderstanding something here?

File: src/sage/knots/link.py [hunk 6]
```diff
@@ -4401,6 +4496,48 @@ def is_isotopic(self, other):
...
+            verbose 1 (... link.py, is_isotopic) identified by KnotInfo (KnotInfo.K7_2, SymmetryMutant.mirror_image)
+            True
+            sage: l1.is_isotopic(l3)
+            verbose 1 (... link.py, is_isotopic) different Homfly-PT polynomials
+            False
+            sage: set_verbose(0)
+
+        TESTS:
+
+        check that :issue:`37668` is fixed::
```
Comment (tscrim) [inline src/sage/knots/link.py]:
```suggestion
        Check that :issue:`37668` is fixed::
```

**Review by tscrim — APPROVED** [review]
LGTM overall; just a few small things.

File: src/sage/knots/knotinfo.py (hunk 1 above)
Comment (soehms) [reply src/sage/knots/knotinfo.py:376]: Yes! I want to have `itself` be the first of the proper items. I will add a comment in the code!

File: src/sage/knots/link.py (hunk 5 above)
Comment (soehms) [reply src/sage/knots/link.py:4398]: The list `l` is merged from the individual lists of different symmetry mutant:

```
        l = []
        if proved and unique and self.is_knot():
            for k in match_lists.keys():
                if proves[k]:
                    l += match_lists[k]
        else:
            # for multi-component links there could regularily be more than one
            # matching entry
            for k in match_lists.keys():
                l += match_lists[k]
```

Therefore, for amphicheiral or reversible links there might be duplicates in it.

Comment (tscrim) [reply src/sage/knots/link.py:4398]: Ah, I see. I misunderstood what the `unique` parameter's purpose is.

**Comment by tscrim** [comment]
Thanks. LGTM.

**Comment by soehms** [comment]
> Thanks. LGTM.

Thanks, too!

**Pull request merged**